components/org.wso2.carbon.identity.api.user.totp/org.wso2.carbon.identity.api.user.totp.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/totp/v1/core/TOTPService.java

```diff
@@ -207,11 +244,12 @@ public TOTPResponseDTO validateTOTP(int verificationCode) {
                             .setKeyRepresentation(encoding);
             TOTPAuthenticatorCredentials totpAuthenticator =
                     new TOTPAuthenticatorCredentials(configBuilder.build());
-            String secretKey = getSecretKey().getSecret();
+            String secretKey = getVerifySecretKey();
```

Comment:
Once the verification is completed, we are deleting VERIFY_SECRET_KEY_CLAIM_URL. So if this validateTOTP method is used for all the TOTP validations, then this getVerifySecretKey should return null right ?

Reply:
The API is used to set/update TOTP from My Account. For TOTP validations in login flows, we are not using these APIs

Reply:
As @bhagyasakalanka mentioned, TOTP validations in login flows will not be handled through this, it will be handled from TOTP AUthenticator side